src/plugins/walletConnectToDapps/eventsManager/useWalletConnectEventsHandler.ts

```diff
@@ -39,7 +39,17 @@ export const useWalletConnectEventsHandler = (
     async (requestEvent: SupportedSessionRequest) => {
       const { topic, params } = requestEvent
       const { request } = params
-      const requestSession = web3wallet?.engine.signClient.session.get(topic)
+      const getRequestSession = () => {
+        try {
+          const allSessions = web3wallet?.engine.signClient.session.getAll()
+          // We should be able to do web3wallet?.engine.signClient.session.get(topic) here, but this always return undefined, another day in closures-land
+          const session = allSessions?.find(s => s.topic === topic)
+          return session
+        } catch (error) {
+          console.error('Failed to get session for topic:', topic, error)
```

Comment:
I did notice this guy throwing when plugging a debugger in, so this adds additional paranoia too